### PR TITLE
Update to wgpu with shader validation

### PR DESCRIPTION
**Pull request by kvark**
Depends on https://github.com/gfx-rs/wgpu/pull/705

### CODE REVIEW

File: examples/mipmap/main.rs [hunk 1]
```diff
@@ -154,10 +154,7 @@ impl Example {
             mag_filter: wgpu::FilterMode::Linear,
             min_filter: wgpu::FilterMode::Nearest,
             mipmap_filter: wgpu::FilterMode::Nearest,
-            lod_min_clamp: 0.0,
-            lod_max_clamp: 100.0,
-            compare: None,
-            anisotropy_clamp: None,
+            .. wgpu::SamplerDescriptor::default()
```
Comment (cwfitzgerald) [inline examples/mipmap/main.rs:157]: This might cause conflicts with #350 as that does this same thing, but less aggressively.

**Comment by cwfitzgerald** [comment]
Now that #350 merged, this should be good to go

File: src/lib.rs [hunk 2]
```diff
@@ -926,9 +926,12 @@ impl Instance {
     pub fn enumerate_adapters(&self, backends: wgt::BackendBit) -> impl Iterator<Item = Adapter> {
```
Comment (cwfitzgerald) [inline src/lib.rs:926]: This should take unsafe extensions as an argument, so the user can choose.

**Comment by kvark** [comment]
Ready now

**Pull request merged**